optuna/visualization/matplotlib/_contour.py

```diff
@@ -175,7 +175,9 @@ def _calculate_axis_data(
     returned_values: Sequence[int | float]
     if axis.is_cat:
         enc = _LabelEncoder()
-        returned_values = enc.fit_transform(list(map(str, values)))
+        # prepare all values in categorical distribution
+        enc.fit(list(map(str, filter(lambda value: value is not None, axis.values))))
```

Comment:
Please note the following comment as well:
- https://github.com/optuna/optuna/pull/5892/files#r1898380046

```suggestion
        # Fit LabelEncoder with all the categories in categorical distribution.
        enc.fit(list(map(str, filter(lambda value: value is not None, axis.values))))
        # Then transform the values using the fitted label encoder.
        # Note that `values` may not include all the categories, so we use `axis.values` for fitting.
```